pkg/controller/cyclenodestatus/transitioner/util.go

```diff
@@ -39,5 +39,11 @@ func (t *CycleNodeStatusTransitioner) transitionObject(desiredPhase v1.CycleNode
 // timedOut returns true if the processing of this CycleNodeStatus has been going longer
 // than the timeout threshold
 func (t *CycleNodeStatusTransitioner) timedOut() bool {
-	return time.Now().After(t.cycleNodeStatus.Status.StartedTimestamp.Time.Add(nodeTerminationGracePeriod))
+	parsedCyclingTimeout, err := time.ParseDuration(t.cycleNodeStatus.Spec.CycleSettings.CyclingTimeout)
```

Comment:
can you parse out this time earlier such that you only have to call it once, and just store the Duration

Reply:
Sounds good! Since this conversion would be called every re-queue